tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/mapper/CurriculumMembershipMapper.java

```diff
@@ -154,6 +172,16 @@ private ProgrammeMembershipDTO curriculumMembershipToProgrammeMembershipDto(
     } else {
       result.setPerson(personToPersonDto(programmeMembership.getPerson()));
     }
+    if (result.getUuid() != null) {
+      Optional<ConditionsOfJoining> conditionsOfJoiningOptional
+          = conditionsOfJoiningRepository.findById(result.getUuid());
+      if (conditionsOfJoiningOptional.isPresent()) {
+        result.setConditionsOfJoining(
+            conditionsOfJoiningMapper.toDto(conditionsOfJoiningOptional.get()));
+      } else {
+        result.setConditionsOfJoining(null);
+      }
```

Comment:
I think the else is unnecessary.  We initialise `result` above.  This means it can be simplified to
```
conditionsOfJoiningOptional.ifPresent(c -> result.setConditionsOfJoining(
    conditionsOfJoiningMapper.toDto(c));
```

or if you want to preserve the explicit set for a null value, use the mapper being null safe:
`result.setConditionsOfJoining(conditionsOfJoiningMapper.toDto(optional.orElse(null)));`

Reply:
That's neater, thanks.